hoomd/operation.py

```diff
@@ -288,7 +288,95 @@ def __getstate__(self):
         return state
 
 
-class Operation(_HOOMDBaseObject):
+class AutotunedObject(_HOOMDBaseObject):
+    """An object with autotuned kernel parameters.
+
+    `AutotunedObject` instances may complete portions of their computation with
+    one or more GPU kernels. Each GPU kernel is executed with a set of
+    parameters (`kernel_parameters`) that influence the run time of the
+    execution. After initialization, `AutotunedObject` varies these parameters
+    as the simulation runs and searches for the best performing combination. The
+    optimal parameters depend on your system's size, density, force field
+    parameters, the specific hardware you execute on, and more.
+
+    Check `is_tuning_complete` to check if the search is complete. After the
+    search is complete, `AutotunedObject` holds the parameters constant.
+    Typical operations require thousands of time steps to complete tuning. Some
+    may take tens of thousands or more depending on the parameters you set.
+
+    Tip:
+        When you significantly change your system during the simulation (e.g.
+        compress to a higher density), then tune the parameters again after
+        making the change with `tune_kernel_parameters`.
+
+    Note:
+        In MPI parallel execution, all methods and attributes of
+        `AutotunedObject` reference the rank local tuner objects. Different
+        ranks may tune different optimal kernel parameters and may complete
+        tuning at different times.
+
```

Comment:
It seems like this has some implications. Could you explain what's implied here?

Reply:
Users can inspect `is_tuning_complete` for individual objects, but should not use them to control the simulation - otherwise deadlock could result. I added a statement indicating that one should use `Operations.is_tuning_complete` for control flow.